src/main/java/uk/gov/justice/digital/hmpps/keyworker/services/GetStaffDetails.kt

```diff
@@ -116,9 +127,9 @@ class GetStaffDetails(
     val prisonName =
       prisonerDetails.values.firstOrNull()?.prisonName ?: prisonRegisterApi.findPrison(prisonCode)!!.prisonName
 
-    val (current, cnSummary) = allAllocations.staffCountStats(fromDate, now(), prisonConfig, staffId)
+    val (current, cnSummary) = allAllocations.staffCountStats(statsReportingPeriod, prisonConfig, staffId)
     val (previous, _) =
-      allAllocations.staffCountStats(previousFromDate, fromDate, prisonConfig, staffId)
+      allAllocations.staffCountStats(statsReportingPeriod.minusMonths(1), prisonConfig, staffId)
```

Comment:
This isn't necessarily the correct decrement any more. If from and to are specified, it should decrement the days between those two rather than a month. Perhaps have a current and previous function in the ReportingPeriod class that calculates this?

Also, with the inclusion of today in the current, will there be an overlap with the end day of previous and the first day of current? I think the ranges used for today would be:

current: 27/05/2025 00:00 -> 28/06/2025 00:00
previous: 27/04/2025 00:00 -> 28/05/2025 00:00